modules/iso14229/server.c

```diff
@@ -1,86 +1,194 @@
 #include <stdint.h>
 #include <stdbool.h>
+#include <string.h>
 #include <zephyr/kernel.h>
+#include <zephyr/sys/atomic.h>
+#include <zephyr/sys/printk.h> // For logging (optional)
+#include <zephyr/kernel_structs.h> // For K_FIFO_DEFINE
 #include <zephyr/logging/log.h>
 #include <zephyr/drivers/can.h>
 #include <zephyr/drivers/gpio.h>
 #include <zephyr/toolchain/common.h>
 #include <zephyr/lib/iso14229/uds.h>
+#include <zephyr/kernel/thread.h>
 
 LOG_MODULE_REGISTER(uds, CONFIG_LOG_DEFAULT_LEVEL);
 
+/* ========================
+ * UDS service handler struct
+ * ======================== */
```

Comment:
Use doxygen notation, same comment goes for rest of the files
```suggestion
/**
 * @brief UDS service handler struct
 */
```